---
layout: paper
title: >-
    Using CRISPRs as a metagenomic tool to identify microbial hosts of a diffuse flow hydrothermal vent viral assemblage: CRISPR spacers reveal hosts of marine vent viral assemblage
image: /images/papers/Anderson et al. - 2011 - Using CRISPRs as a metagenomic tool to identify mi.png
authors: Anderson, Rika E.; Brazelton, William J.; Baross, John A.
year: 2011
ref: Anderson et al. 2011 FEMS Microbiol Ecol
journal: >-
    FEMS Microbiology Ecology
pdf: /pdfs/papers/Anderson et al. - 2011 - Using CRISPRs as a metagenomic tool to identify mi.pdf
supplement: 
doi: 10.1111/j.1574-6941.2011.01090.x
---

Link to online paper: [https://academic.oup.com/femsec/article-lookup/doi/10.1111/j.1574-6941.2011.01090.x](https://academic.oup.com/femsec/article-lookup/doi/10.1111/j.1574-6941.2011.01090.x)

# Abstract

Metagenomic analyses of viruses have revealed widespread diversity in the viriosphere, but it remains a challenge to identify speciﬁc hosts for a viral assemblage. To address this problem, we analyze the viral metagenome of a northeast Paciﬁc hydrothermal vent with a comprehensive database of spacers derived from the clustered regularly interspaced short palindromic repeat (CRISPR) putative immune system. CRISPR spacer matches to the marine vent virome suggest that viruses infecting hosts from diverse taxonomic groups are present in this vent environment. Comparative virome analyses show that CRISPR spacers from vent isolates and from thermophiles in general have a higher percentage of matches to the vent virome than to other marine or terrestrial hot spring viromes. However, a high percentage of hits to spacers from mesophilic hosts, combined with a moderately high modeled alpha diversity, suggest that the marine vent virome is comprised of viruses that have the potential to infect diverse taxonomic groups of multiple thermal regimes in both the bacterial and the archaeal domains.

